lib/puppet/provider/x509_request/openssl.rb

```diff
@@ -38,21 +38,17 @@ def exists?
   end
 
   def create
-    cmd_args = [
+    options = [
       'req', '-new',
       '-key', resource[:private_key],
       '-config', resource[:template],
       '-out', resource[:path]
     ]
 
-    if resource[:password]
-      cmd_args.push('-passin')
-      cmd_args.push("pass:#{resource[:password]}")
-    end
-
-    cmd_args.push('-nodes') unless resource[:encrypted]
+    options << ['-passin', "pass:#{resource[:password]}"] if resource[:password]
```

Comment:
This add a single item (an array with two elements)

Reply:
https://github.com/voxpupuli/puppet-openssl/pull/155#discussion_r1122812489